### ISSUE TITLE
Can't build next branch in VS2017

### ISSUE THREAD

**Issue by afalturki** [gcp/leela-zero#828]
Hi,

I'm trying to build the next branch with the latest changes in Visual Studio 2017 but I'm getting 8 errors for autogtp all about unresolved external symbols.

For example, this is one error:

Severity	Code	Description	Project	File	Line	Suppression State
Error	LNK2001	unresolved external symbol "public: void __cdecl Order::load(class QString const &)" (?load@Order@@QEAAXAEBVQString@@@Z)	autogtp	C:\Users\afalt\leela-zero\msvc\VS2017\Management.obj	1	

I'm not sure if the latest changes broke compatibility with VS or I'm doing something wrong.

**Comment by odeint** [comment]
Broke for me too, but just autogtp. You can just compile the LeelaZero part of the project only and use it with an earlier autogtp. All the important changes have been to LZ, not autogtp.

**Comment by afalturki** [comment]
Yeah. Same here. Just autogtp is broken.

**Comment by CheckersGuy** [comment]
are there any plans  when the merge with next is going to happen ?

**Comment by marcocalignano** [comment]
In autogtp there is the new "stop-resume" feature. This feature introduced a new file Order.c that wasn't there before. I bet the VS2017 buildfile is missing the command to compile and link it. 
Now I do not have VS2015 or VS2017 and I do not know how to do it, but I am sure someone out there can fix this problem.

**Comment by afalturki** [comment]
@marcocalignano I see. ty. I'll try to add it manually until a new commit fixes the problem.

**Comment by gcp** [comment]
>are there any plans when the merge with next is going to happen ?

I want to pull the improvements in the "tune first play urgency" thread for near-uniform networks. Testing is halfway, so "soon".

**Comment by marcocalignano** [comment]
Please post the errors text so I can have a look.

**Comment by afalturki** [comment]
They’re the same errors @odeint posted.

**Comment by marcocalignano** [comment]
Yes there is also Console.h that is new. And even if you do not need to compile it because is included in main.cpp it is a QObject and it need it moc file for signal vtable.

**Comment by afalturki** [comment]
Ok I’ll play around and try to add them. Is the order important?

**Comment by marcocalignano** [comment]
I don't think so, I think linking has to be at end! :smile:

**Comment by odeint** [comment]
I think I know how to do it, but I can only try it tonight at home. Moc'ing was definitely working for me, I'll just have to add the extra header and cpp files. Should have checked the project file against the contents of the folder :-)

**Comment by afalturki** [comment]
I'm not really a programmer, so, I'm probably doing something stupid here. I tried adding the missing files but I'm still getting 6 errors. Could you please take a look at what I'm doing wrong? Here's what I did:

https://github.com/afalturki/Leela-Zero-MSVC/pull/1/files

**Comment by odeint** [comment]
> Could you please take a look at what I'm doing wrong? Here's what I did:

Hi @afalturki, I wouldn't actually directly edit those MSVC XML files, that's not the VisualStudio way. In the GUI, expand the tree in the project view. Then add the missing files there, or to be sure just remove them all first and then add all the .cpp and .h files in the source and header sections. Don't have VS here so I can't show you a screenshot, but the gist is to do it in VS and not edit the files directly.

**Comment by afalturki** [comment]
I see. Thanks. I'll try that

**Comment by afalturki** [comment]
Thank you very much. It worked!

**Comment by Cabu** [comment]
The last patch solve the problem! Thanks!

**Comment by afalturki** [comment]
@Cabu You're welcome! Glad to help (:
